app/src/main/java/com/github/se/travelpouch/ui/authentication/SignIn.kt

```diff
@@ -196,14 +205,19 @@ fun SignInScreen(
           }
           if (currentUser != null &&
               !signInCompleted
-                  .value) { // extra condition on sign-in completion to avoid unnecessary
+                  .value && isConnectedToNetwork.value != null) { // extra condition on sign-in completion to avoid unnecessary
             // recompositions
             LaunchedEffect(Unit) {
               try {
                 isLoading.value = true
                 methodChosen.value = true
 
-                currentUser.getIdToken(true).await() // We shouldn't continue until this passes
+                if (isConnectedToNetwork.value == true) {
```

Comment:
Are you sure that you have to test the something is equat to true ?

Reply:
Yes I am, keep in mind it's not a Boolean but a Boolean? where the null state represents the helper in the initial state (very short but still a case to handle)